### PR TITLE
Add path-based fingerprint override for method/class offenses

### PR DESCRIPTION
**Pull request by dblandin**
This PR implements override fingerprints for method and class-based offenses, incorporating the path, check name, and offense message.

@codeclimate/review :mag_right:

### CODE REVIEW

**Comment by wfleming** [comment]
Do I understand correctly that these fingerprints would be identical to the current default?

I'd like to dig a bit more into the monkey patching before we bail entirely on that. Maybe a good thing to pair on.

**Comment by dblandin** [comment]
> Do I understand correctly that these fingerprints would be identical to the current default?

That's correct.

> I'd like to dig a bit more into the monkey patching before we bail entirely on that. Maybe a good thing to pair on.

Sounds good to me!

**Comment by dblandin** [comment]
@wfleming Ready for another :mag_right:

File: bin/codeclimate-rubocop [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env ruby
-$LOAD_PATH.unshift(
+$:.unshift(
```
Comment (wfleming) [inline bin/codeclimate-rubocop]: Unrelated change? Doid rubocop start complaining about this?

Comment (dblandin) [inline bin/codeclimate-rubocop]: Yep, this was recommended by RuboCop in this [rule](http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/SpecialGlobalVars).

File: spec/cc/engine/issue_spec.rb [hunk 2]
```diff
@@ -36,6 +36,14 @@ module CC::Engine
           "This cop checks that the cyclomatic complexity of methods is not higher than the configured maximum."
         )
       end
+
+      it "sets a fingerprint for method/class offenses" do
```
Comment (wfleming) [inline spec/cc/engine/issue_spec.rb]: Worth asserting that it does *not* set a fingerprint for some other cop it shouldn't?

File: bin/codeclimate-rubocop (hunk 1 above)
Comment (wfleming) [inline bin/codeclimate-rubocop]: Oh, it's the new base rubocop. nevermind.

File: lib/cc/engine/fingerprint.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          md5 << stripped_message
+          md5.hexdigest
+        end
+      end
+
+      private
+
+      attr_reader :path, :cop_name, :message
+
+      def check_name
```
Comment (wfleming) [inline lib/cc/engine/fingerprint.rb]: nit, but there's small duplication between this and `Issue#check_name`. And this one doesn't really need the `Rubocop/` bit, since we're just hashing it. Thoughts on using `cop_name` directly in  the hash instead here?

File: spec/cc/engine/issue_spec.rb (hunk 2 above)
Comment (dblandin) [inline spec/cc/engine/issue_spec.rb]: Sounds good. Added 👍

File: lib/cc/engine/fingerprint.rb (hunk 3 above)
Comment (dblandin) [inline lib/cc/engine/fingerprint.rb]: Sounds good to me. Updated 👍

**Comment by wfleming** [comment]
LGTM
